Automattic/wp-calypso
# WooCommerce: Show momma when store has orders with process orders widget

I closed the WIP branch at #15332 in favor of this PR. Also see #14205

This PR finishes implementing the logic of when to show the Momma view of the dashboard, and implements the "Process Orders" widget if there are new orders to show. I added some additional helper selectors to get just "new" orders, and a combined amount of their total fields. 

Orders currently being considered new: `pending`, `processing` and `on-hold`. Basically orders that require an action on the part of the store manager.

<img width="876" alt="screen shot 2017-06-22 at 8 53 34 am" src="https://user-images.githubusercontent.com/689165/27445064-6208578a-572d-11e7-8043-05c9fb6ffd05.png">

To Test:
* Make sure you have finished setup via the dashboard.
* Make sure you have some orders to display on your site. Have one or two of these marked as `completed` and the rest `pending`, `processing`, or `on-hold`.
* Go to `http://calypso.localhost:3000/store/:site`
* You should see a version of the dashboard welcoming you back (with your correct display name or username)
* You should see the processing widget. It should include the # of new orders and a total amount. Verify the "Process orders" button takes you to the order list.
* In `wp-admin`, Trash your `pending`, `processing`, or `on-hold` orders (or mark them as completed) and refresh the dashboard. The widget should go away. You should still see the welcome message and a button to view reports.

cc @allendav 
cc @kellychoffman @jameskoster since this contains UI.
